src/swift/nullObject/Sources/nullObject/loginFactory.swift

```diff
@@ -0,0 +1,16 @@
+public struct LoginFactory {
+    
+    public enum ServiceType: String {
+        case facebook, google
+    }
+
+    private let loginServices: [String: () -> LoginService] = [
+        ServiceType.facebook.rawValue: { FacebookLogin() },
+        ServiceType.google.rawValue: { GoogleLogin() },
+    ]
+
+    func getService(ofType type: String) -> LoginService {
+        return loginServices[type]?() ?? NullLogin()
```

Comment:
You can consider to use the new swift syntax that allow you to specify a default value in case of dictionary key not present.
For example:
```
var arr = myDict["key", default: []]
```